.github/workflows/trigger-e2e-on-comment.yml

```diff
@@ -22,23 +22,16 @@ jobs:
       - name: Run E2E Tests
         run: |
           # Github does not provide the branch of a PR when a comment on a PR is made
-          # So, given the issue number, which is the PR number, we can retrieve the branch with
-          # a quick API call
-          echo "Retrieving branch"
-          BRANCH=$(curl -L \
-            -H "Accept: application/vnd.github+json" \
-            -H "Authorization: Bearer $GITHUB_TOKEN" \
-            -H "X-GitHub-Api-Version: 2022-11-28" \
-            https://api.github.com/repos/facebook/react-native/pulls/$PR_NUMBER | jq -r '.head.ref')
-
-          echo "Trigger Test All workflow for branch $BRANCH"
+          # So, given the issue number, which is the PR number, we can use GitHub's refs. E.g. refs/pull/1/merge
+          echo "Trigger Test All workflow for PR $PR_NUMBER"
+          
           curl -L \
             -X POST \
             -H "Accept: application/vnd.github+json" \
             -H "Authorization: Bearer $GITHUB_TOKEN" \
             -H "X-GitHub-Api-Version: 2022-11-28" \
             https://api.github.com/repos/facebook/react-native/actions/workflows/test-all.yml/dispatches \
-            -d "{\"ref\": \"$BRANCH\", \"inputs\": {\"run-e2e-tests\": \"true\"}}"
+            -d "{\"ref\": \"refs/pull/$PR_NUMBER/merge\", \"inputs\": {\"run-e2e-tests\": \"true\"}}"
```

Comment:
Can't get just use `GITHUB_REF` here instead?

Reply:
Don't think so. `GITHUB_REF` at `issue_comment` events will use `default branch`, but this trigger should use the branch of the PR the comment was added to.

![image](https://github.com/user-attachments/assets/2d3a5924-13d0-48d1-9020-b0f804426aeb)
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#issue_comment